frontend/src/pages/RunListsRouter.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import { RunStorageState } from 'src/apis/run';
+import MD2Tabs from 'src/atoms/MD2Tabs';
+import { commonCss, padding } from 'src/Css';
+import { classes } from 'typestyle';
+import RunList, { RunListProps } from './RunList';
+
+export enum RunListsGroupTab {
+  ACTIVE = 0,
+  ARCHIVE = 1,
+}
+
+export type RunListsRouterProps = RunListProps & {
+  storageState: RunStorageState;
+  onTabSwitch?: (tab: RunListsGroupTab) => void;
+};
+
+/**
+ * Contains two tab buttons which allows user to see Active or Archived runs list.
+ */
+class RunListsRouter extends React.PureComponent<RunListsRouterProps> {
+  protected _isMounted = true;
+  private _runlistRef = React.createRef<RunList>();
```

Comment:
@Bobgy this is a great argument! Do you think you could start an issue to document these antipatterns, maybe with a brief proposed initial solution, so we can keep track of them? You know the repository and the codebase better than anyone else. If we can keep our context there and start a discussion, then we will be able to address these one by one

Reply:
Thank you @Bobgy and @StefanoFioravanzo ! I agree with following one-way data flow approach and we should avoid using Ref as much as we can. I have attempted to change my PR to follow the suggested guide. But I found out that it requires changes beyond just `ExperimentDetail.tsx`, therefore I cautiously replace the implementation in `ExperimentDetail.tsx` with the one-way data flow practice, while keeping other files with Ref approach for now. `RunList.tsx` change affects multiple pages so I leave it unchanged to avoid large PR. We can take it as a first step of migration away from Ref, and gradually refactoring the codebase along the way.

To @StefanoFioravanzo: Good suggestion! I have created https://github.com/kubeflow/pipelines/issues/5118 so feel free to make your proposal there. The goal is to document the agreed practices in README file so all contributors can follow along.

Reply:
Awesome, I'll put more there soon